src/nbl/asset/utils/IShaderCompiler.cpp

```diff
@@ -171,7 +101,7 @@ namespace nbl::asset::impl
 
 std::string IShaderCompiler::preprocessShader(
     std::string&& code,
-    IShader::E_SHADER_STAGE stage,
+    IShader::E_SHADER_STAGE& stage,
```

Comment:
Ok so I know whats goin on, hoist all of this to CGLSLCompiler and make this IShaderCompiler function a pure virtual one.

Reply:
These are all glsl specific so try to move all of the preprocessing shit to its class